features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF5/source/nRF5xPalSecurityManager.cpp

```diff
@@ -727,9 +792,32 @@ bool nRF5xSecurityManager::sm_handler(const ble_evt_t *evt)
             return true;
         }
 
-        case BLE_GAP_EVT_LESC_DHKEY_REQUEST:
-            // FIXME: Add with LESC support
+        case BLE_GAP_EVT_LESC_DHKEY_REQUEST: {
+            const ble_gap_evt_lesc_dhkey_request_t& dhkey_request =
+                gap_evt.params.lesc_dhkey_request;
+
+            size_t key_size = public_key_coord_t::size();
+            public_key_coord_t peer_X(dhkey_request.p_pk_peer->pk, key_size);
+            public_key_coord_t peer_Y(dhkey_request.p_pk_peer->pk + key_size, key_size);
```

Comment:
you don't need to pass in the size to the constructor and what's wrong with sizeof?